ouroboros-consensus/test-consensus/Test/Consensus/HardFork/Combinator.hs

```diff
@@ -82,59 +84,81 @@ tests = testGroup "HardForkCombinator" [
     ]
 
 data TestSetup = TestSetup {
-      testSetupSeed :: Seed
+      -- TODO: Vary epoch size across the fork
+      testSetupEpochSize  :: EpochSize
+      -- ^ INVARIANT: @> 0@
+    , testSetupK          :: SecurityParam
+    , testSetupSeed       :: Seed
+      -- TODO: Vary slot length across the fork
+    , testSetupSlotLength :: SlotLength
+    , testSetupTxSlot     :: SlotNo
     }
   deriving (Show)
 
 instance Arbitrary TestSetup where
   arbitrary = do
-      testSetupSeed <- arbitrary
+      testSetupEpochSize <- EpochSize     <$> choose (1, 10)
+      testSetupK         <- SecurityParam <$> choose (2, 10)
+      -- TODO why does k=1 cause the nodes to only forge in the first epoch?
```

Comment:
I'll check if this seems related to Issue #1511, which is overdue for a revisit since so much consensus logic has changed since I last tried.